### DISCUSSION TITLE
WSL2 not loading custom kernel

### DISCUSSION THREAD

**Discussion by bmaciag** [microsoft/WSL#6156] [General] [answered]
# Environment

```none
Windows build number: 10.0.20241.1005
Your Distribution version: 20.04
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.4.0-20241-Microsoft (Microsoft@Microsoft.com) (gcc version 5.4.0 (GCC) ) #1000-Microsoft Thu Oct 16 14:50:00 PST 2020
```

# Steps to reproduce

Download the latest kernel release: https://github.com/microsoft/WSL2-Linux-Kernel/releases/tag/4.19.128-microsoft-standard
Build and use via `.wslconfig`: 
```
[wsl2]
kernel=C:\\Users\\martiu\\bzImage
```

**WSL logs**: 

#  Expected behavior

`uname -ar` lists the kernel wich i build, instead I see the default one.

# Actual behavior

`4.4.0-20241-Microsoft from Oct 16th` and not my kernel which I build.

**Comment by Biswa96** [comment]
Use the `vmlinux` file.

**Comment by bmaciag** [comment]
If I use the `arch/x86/boot/vmlinux.bin` instead of `arch/x86/boot/bzImage` its the same result, its not loading.

EDIT: Even if I take the `vmlinux` from the build from github, its the same

**Comment by 0xbadfca11** [comment, marked as answer]
``4.4.0-20241-Microsoft`` is WSL 1.
You are not using WSL 2.

**Comment by Barshan-Mandal** [comment]
But I am using wsl2.Please help me
